src/signature.c

```diff
@@ -112,6 +112,7 @@ void terminate_signature(void)
 		fclose(fp_pubkey);
 	}
 	if (cert) {
+		X509_free(cert);
```

Comment:
Running into a segfault now with this line added... My previous testing showed that this plugged a memory leak, but now I'm seeing invalid frees being reported.

I will remove this line.